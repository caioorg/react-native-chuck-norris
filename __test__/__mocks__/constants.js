export const LIST_CATEGORY = [
  'animal',
  'career',
  'celebrity',
  'dev',
  'explicit',
  'fashion',
  'food',
  'history',
  'money',
  'movie',
  'music',
  'political',
  'religion',
  'sport',
  'travel',
];

export const JOKE_SELECTED = {
  value:
    'All arrays Chuck Norris declares are of infinite size, because Chuck Norris knows no bounds.',
  categories: ['dev'],
};
