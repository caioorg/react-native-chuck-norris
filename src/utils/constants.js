export const URLS = {
	CATEGORIES: 'https://api.chucknorris.io/jokes/categories',
	RANDOM_CATEGORY_FACT: 'https://api.chucknorris.io/jokes/random?category=',
};
