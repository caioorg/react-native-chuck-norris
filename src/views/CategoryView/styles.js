import styled from 'styled-components/native';

export const ItemList = styled.TouchableOpacity`
  padding: 20px 10px;
`;

export const ItemListText = styled.Text`
  font-size: 20px;
  text-transform: capitalize;
`;
