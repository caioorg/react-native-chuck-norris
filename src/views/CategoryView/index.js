import React from 'react';
import PropTypes from 'prop-types';
import {View, FlatList} from 'react-native';
import {ItemList, ItemListText} from './styles';

const CategoryView = ({data, onSelecteCategory}) => {
  return (
    <View>
      <FlatList
        data={data}
        renderItem={({item}) => {
          return (
            <ItemList onPress={() => onSelecteCategory(item)} key={item}>
              <ItemListText>{item}</ItemListText>
            </ItemList>
          );
        }}
      />
    </View>
  );
};

CategoryView.propTypes = {
  data: PropTypes.array.isRequired,
  onSelecteCategory: PropTypes.func.isRequired,
};

export default CategoryView;
