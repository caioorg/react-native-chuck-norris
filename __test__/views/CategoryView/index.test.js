import React from 'react';
import renderer from 'react-test-renderer';
import CategoryView from '../../../src/views/CategoryView';
import {LIST_CATEGORY} from '../../__mocks__/constants';

const props = {data: LIST_CATEGORY, onSelecteCategory: jest.fn()};

describe('[View] Category', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<CategoryView {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
