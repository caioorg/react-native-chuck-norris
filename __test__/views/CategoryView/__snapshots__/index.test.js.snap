// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        "animal",
        "career",
        "celebrity",
        "dev",
        "explicit",
        "fashion",
        "food",
        "history",
        "money",
        "movie",
        "music",
        "political",
        "religion",
        "sport",
        "travel",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            animal
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            career
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            celebrity
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            dev
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            explicit
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            fashion
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            food
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            history
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            money
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            movie
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
